### DISCUSSION TITLE
Aligning internal GraalVM versioning with JDK versions

### DISCUSSION THREAD

**Discussion by alina-yur** [oracle/graal#9480] [General]
At the moment, GraalVM uses both JDK versioning (GraalVM for JDK 22, GraalVM for JDK 23, ...) and our internal versioning (GraalVM for JDK 22 = GraalVM v24.0, GraalVM for JDK 23 = v24.1). To better align with JDK releases, and also to avoid confusion, we will slow down our versioning for one release, so that we can catch up with JDK versioning starting with JDK 25. 

The change will work as follows:

| GraalVM version | Java version |
|-----------------|--------------|
| 24.1            | 23           |
| 24.2            | 24           |
| 25.0            | 25           |
| 26.0            | 26           |

So the key change is that GraalVM for JDK 24 will have internal version 24.2, and starting with GraalVM for JDK 25 in September 2025 the versions will be fully aligned. The new versioning will also apply to Truffle and Graal languages.

If you are integrating with GraalVM, or tracking internal versions, please adjust for this change.

**Comment by haribabupuppala** [comment]
Could you please share the earlier versions of Graal VM and its corresponding JDK version compatibility?.

**Reply by alina-yur** [reply]
hi @haribabupuppala. are there any specific GraalVM versions you are interested in?

**Reply by haribabupuppala** [reply]
We are currently using Graal VM with JDK 21 for which we need the JDK compatibility matrix
I could not find the official oracle documentation, which shows the compatibility matrix for Graal VM and its supported JDK versions

**Reply by alina-yur** [reply]
This page goes back through all our major releases, I hope it helps: https://www.graalvm.org/release-calendar/? Otherwise you can give me an internal GraalVM version your are interested in. For GraalVM for JDK 21, it's built for JDK 21, but it's keeps receiving CPU updates (Oracle GraalVM).

**Reply by haribabupuppala** [reply]
Thank you for sharing the above information, but I am looking for the version mapping between graal  and java version as mentioned in your initial post in tabular format,
